### PR TITLE
cdhit dagly webs cleanup

### PR DESCRIPTION
**Pull request by boris-dimitrov**
Post-deployment cleanup for the CD-HIT-DUP work.   Pre-requisite to https://github.com/chanzuckerberg/idseq-dag/pull/268.   Based on Greg's initial draft PR.

It does two things:

1) Emit orrect DAG

2) Tolerate additional attributes in [refined_]taxon_counts_with_dcr.json

Please see comments below for test plan and post-deployment.

### CODE REVIEW

File: app/lib/dags/postprocess.json.jbuilder [hunk 1]
```diff
@@ -37,19 +37,19 @@ json.targets do
...
+    "assembly/refined_gsnap_counts_with_dcr.json",
     "assembly/gsnap_contig_summary.json",
     "assembly/gsnap.blast.top.m8",
   ]
   json.refined_rapsearch2_out [
     "assembly/rapsearch2.blast.m8",
     "assembly/rapsearch2.reassigned.m8",
     "assembly/rapsearch2.hitsummary2.tab",
-    "assembly/refined_rapsearch2_counts.json",
+    "assembly/refined_rapsearch2_counts_with_json.json",
```
Comment (gregdingle) [inline app/lib/dags/postprocess.json.jbuilder]: Should this be with_dcr.json?

Comment (boris-dimitrov) [reply app/lib/dags/postprocess.json.jbuilder]: Yes, good catch.

**Comment by boris-dimitrov** [comment]
Copying from the older PR the comment of Charles De Bourcy which is a very helpful suggestion for testing this:

Let's get this change ready for deployment by doing the testing already. You can do the end-to-end testing change by submitting a sample from your local environment (and scaling up staging alignment machines). You can upload the sample from localhost:3000/samples/new (legacy interface) and specify a branch of idseq-dag to be used for processing, which should essentially be https://github.com/chanzuckerberg/idseq-dag/pull/252/files minus the hacks you're seeking to remove.

**Comment by boris-dimitrov** [comment]
Please test as follows, using the local env test instructions from Charles De Bourcy's comment above.  You will *not* need to make any changes to idseq-dag for the initial test.

1) The currently deployed idseq-dag was designed to work equally well with or without this PR.

If that isn't true, change this PR or the deployed idseq-dag or both as needed --- but without deploying anything to idseq-web, yet

Only then proceed to step 2.

2) Test this PR with idseq-dag branch that sets count.DAG_SURGERY_HACKS_FOR_READ_COUNTING=False.   If that works, merge this PR into master, and deploy it at will.  Don't make any changes to idseq-dag as part of step 2.

3) After this PR is deployed, and a few days of happyness have elapsed, remove from idseq-dag all the code that becomes dead when count.DAG_SURGERY_HACKS_FOR_READ_COUNTING=False.

**Comment by boris-dimitrov** [comment]
Here is the idseq-dag PR that you can merge after this idseq-web PR has been happily deployed to production:  https://github.com/chanzuckerberg/idseq-dag/pull/268

**Comment by boris-dimitrov** [comment]
In testing this please consider 3 different test cases:

1)  A tiny sample with human host.
2)  A tiny mosquito sample.
3) A small sample with unpaired reads.

If it works for all 3, it's good.

**Comment by gregdingle** [comment]
> Test this PR with idseq-dag branch that sets count.DAG_SURGERY_HACKS_FOR_READ_COUNTING=False. 

I did that for all of 

> A tiny sample with human host.
> A tiny mosquito sample.
> A small sample with unpaired reads.

And with idseq-dag master branch for good measure.

I'm going to merge now and proceed to 

> After this PR is deployed, and a few days of happyness have elapsed, remove from idseq-dag all the code that becomes dead when count.DAG_SURGERY_HACKS_FOR_READ_COUNTING=False.

**Pull request merged**